src/charm.py

```diff
@@ -629,6 +629,19 @@ def _external_url(self) -> str:
         """Return the externally-reachable (public) address of the alertmanager api server."""
         return self.model.config.get("web_external_url") or self.ingress.url or self._internal_url
 
+    @property
+    def _alertmanager_version(self) -> str:
+        """Returns the version of alertmanager.
+
+        Returns:
+            A string equal to the agent version.
+        """
+        container = self.unit.get_container(self._container_name)
+        if not container.can_connect():
+            return None
```

Comment:
In that case the return type is `Optional[str]`. Odd if mypy didn't catch that.

Reply:
Fixed.